modules/auxiliary/admin/kerberos/ktutil.rb

```diff
@@ -0,0 +1,103 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Kerberos keytab utilities',
+        'Description' => %q{
+          Utilities for interacting with MIT keytab files, which can store the hashed passwords of one or
+          more principals.
+
+          Discovered keytab files can be used to generate Kerberos Ticket Granting Tickets, or bruteforced
+          offline through offline brute forcing attacks.
+
+          Keytab files can be also useful for decrypting Kerberos traffic using Wireshark dissectors,
+          including the krbtgt encrypted blobs if the AES password hash is used.
+        },
+        'Author' => [
+          'alanfoster' # Metasploit Module
+        ],
+        'References' => [
+        ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [],
+          'SideEffects' => [],
+          'Reliability' => [],
+          'AKA' => []
+        },
+        'Actions' => [
+          ['LIST', { 'Description' => 'View the entries in the keytab file' }],
+        ],
+        'DefaultAction' => 'LIST'
+      )
+    )
+
+    register_options(
+      [
+        OptPath.new('KEYTAB_FILE', [false, 'The keytab file to manipulate']),
+        OptEnum.new('OUTPUT_FORMAT', [true, 'The output format to use', 'table', %w[csv table]]),
+      ]
+    )
+  end
+
+  def run
+    case action.name
+    when 'LIST'
+      list_keytab_entries
+    end
+  end
+
+  def list_keytab_entries
+    if datastore['KEYTAB_FILE'].blank? || !File.exist?(datastore['KEYTAB_FILE'])
+      fail_with(Failure::BadConfig, 'Invalid key tab file')
+    end
+
+    tbl = Rex::Text::Table.new(
+      'Header' => 'Keytab entries',
+      'Indent' => 1,
+      'WordWrap' => false,
+      'Columns' => %w[
+        vno
+        type
+        principal
+        hash
+        date
+      ]
+    )
+
+    keytab = File.binread(datastore['KEYTAB_FILE'])
+    keytab = Rex::Proto::Kerberos::KeyTab::KeyTab.read(keytab)
```

Comment:
This `KeyTab` repetition is somewhat upsetting; Alternative suggestions appreciated